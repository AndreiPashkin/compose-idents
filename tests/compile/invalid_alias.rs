use compose_idents::compose_idents;

compose_idents!(alias = lower(42), {
    fn alias() -> u32 {
        42
    }
});

fn main() {
    assert_eq!(qux_foo_baz(), 42);
}
