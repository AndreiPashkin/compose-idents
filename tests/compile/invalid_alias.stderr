error: Invalid alias value: 42
 --> tests/compile/invalid_alias.rs:3:25
  |
3 | compose_idents!(alias = lower(42), {
  |                         ^^^^^

error[E0425]: cannot find function `qux_foo_baz` in this scope
  --> tests/compile/invalid_alias.rs:10:16
   |
10 |     assert_eq!(qux_foo_baz(), 42);
   |                ^^^^^^^^^^^ not found in this scope
